### DISCUSSION TITLE
Jellyfish ve Undertale (codeforces ödevi)

### DISCUSSION THREAD

**Discussion by PurpleResul** [sonsuzus/sonsuzus#527] [Soru Cevap]
Flowey, Snowdin'de bir bomba yerleştirdi!

Bomba, başlangıçta b olarak ayarlanan bir zamanlayıcıya sahip. Her saniye, zamanlayıcı 1 azalır. Zamanlayıcı 0'a ulaştığında, bomba patlar! Snowdin sakinlerinin kaçmak için yeterli zamanı olabilmesi için, bombanın patlamasını mümkün olduğunca geciktirmeniz gerekiyor.

n tane aracınız var. Her araç yalnızca bir kez kullanılabilir. Eğer i-inci aracı kullanırsanız, zamanlayıcı xi kadar artar. Ancak, zamanlayıcı a'dan daha büyük bir değere ulaşırsa, bir hata nedeniyle zamanlayıcı a'ya sabitlenir.

Her saniye aşağıdaki işlemler sırasıyla gerçekleşir:

Kullanılmamış araçlardan bazılarını seçebilirsiniz. Eğer i-inci aracı seçerseniz, ve bombanın zamanlayıcısı şu anda c ise, zamanlayıcı min(c+xi, a) olarak değiştirilir.
Zamanlayıcı 1 azalır.
Eğer zamanlayıcı 0'a ulaşırsa, bomba patlar.
Jellyfish, araçlar optimal şekilde kullanıldığında, bombanın patlamasına kadar geçen maksimum süreyi hesaplamak istiyor.

Girdi:

Birden fazla test vakası içerir. İlk satırda test vakası sayısı t (1≤t≤2000) verilir. Test vakalarının açıklamaları şu şekildedir:

İlk satırda, üç tam sayı a, b ve n (1≤b≤a≤10^9, 1≤n≤100) verilir — bombanın zamanlayıcısının maksimum değeri, bombanın başlangıç zamanı ve araç sayısı.
İkinci satırda, n tane tam sayı x1, x2, ..., xn (1≤xi≤10^9) verilir — araçların zamanlayıcıyı ne kadar artırabileceği.
Çıktı:

Her test vakası için, bombanın patlamasına kadar geçen maksimum süreyi bir tam sayı olarak yazdırın.

not: Undertale çok güzel bir oyun.

**Comment by PurpleResul** [comment]
`t = int(input())
for _ in range(t):
    a, b, n = input().split()
    a, b, n = int(a), int(b), int(n)
    aletler = []
    for _ in range(n):
        aletler.append(int(input()))
    aletler.sort(reverse=True)
    time = 0
    for x in aletler:
        if b == 0:
           break
        b = min(b + x, a) - 1
        time += 1
    print(time + b)
`

**Reply by PurpleResul** [reply]
Hocam kazayla düz yazı gibi yazdı ama birşey olmaz değil mi?

**Reply by sonsuzus** [reply]
Elbette bir şey olur. Düzeltip at tekrar.

**Comment by Yagiz10** [comment]
```
      💙         🩵

💜        ❤️          💚

      💛         🧡
```

**Reply by PurpleResul** [reply]
:)

**Comment by PurpleResul** [comment]
/
```python

t = int(input())
for _ in range(t):
     a, b, n = input().split() a, b, n = int(a), int(b), int(n) 
    aletler = [] 
    for _ in range(n):
        aletler.append(int(input())) aletler.sort(reverse=True) 
        time = 0 
        for x in aletler:
            if b == 0: 
                break b = min(b + x, a) - 1 
                time += 1 
                print(time + b)
